app/controllers/application_controller.rb

```diff
@@ -22,7 +22,7 @@ def set_unit_and_membership
     if user_signed_in?
       current_user.membership = session[:user_memberships]
       if is_super_admin?(current_user)
-        current_user.unit = Unit.all
+        current_user.unit = Unit.all.pluck(:id)
```

Comment:
I think with this change, it will be better if you rename _**current_user.unit**_ to _**current_user.unit_ids**_

Reply:
I will create a separate branch to rename this variable because the renaming will affect all controllers and many views.
And they all should be tested.